File: Unidad_1/ej1.py
# 1) Genere un listor de 10 números complejos, calcule el módulo y aplique un redondeo al
# entero más próximo. El listor resultante tendrá números enteros positivos.
#   a) Separe los valores impares de los pares en diferentes listores.
#   b) Ordene de menor a mayor los elementos de ambos listores.

import cmath
import numpy as np

listComplex = []
listModule = []
listPar = []
listImpar = []

for _ in range(0, 10):
    real = np.random.uniform(0, 10)
    comp = np.random.uniform(0, 10)
    z = real + 1j*comp
    listComplex.append(z)

for c in listComplex:
    listModule.append(round(abs(c)))

for m in listModule:
    r = m % 2
    if r == 0:
        listPar.append(m)
    else:
        listImpar.append(m)

listPar.sort()
listImpar.sort()

print(f'Lista de complejos: {listComplex}')
print(f'Lista de modulos redondeados {listModule}')
print(f'Lista de pares: {listPar}')
print(f'Lista de impares: {listImpar}')

File: matlab/Unidad_1/ej1.m
% 1) Genere un vector de 10 números complejos, calcule el módulo y aplique
% un redondeo al entero más próximo. El vector resultante tendrá números
% enteros positivos.
%   a) Separe los valores impares de los pares en diferentes vectores.
%   b) Ordene de menor a mayor los elementos de ambos vectores.

large = 10;
factor = 10;
v = zeros(1, large);
v_i = [];
v_p = [];

for i = 1:large
    cvalue = complex(rand(1)*factor, rand(1)*factor); % Complejo
    value = round(abs(cvalue)); % Modulo y Redondeo
    v(i) = value; % Guardo en vector v
    if mod(value, 2) == 0
        v_p = [v_p, value];
    else
        v_i = [v_i, value];
    end
end

v_p = sort(v_p);
v_i = sort(v_i);

disp(v);
disp(v_p);
disp(v_i);